README.md

```diff
@@ -190,6 +190,49 @@ Three macros are provided for adding benchmark templates.
 #define BENCHMARK_TEMPLATE2(func, arg1, arg2)
 ```
 
+### Batched processing benchmarks
+
+Sometimes you want to benchmark a batch operation and compare performance for
+different values of batch size. For example, say we have some
+`ParameterizedRoutine` and we want to compare performance as a function of
+the parameter value and not the number of function calls. This could be done
+with:
+
+```c++
+static void BM_ParameterizedRoutine(benchmark::State& state) {
+  const int param = state.range(0);
+  while (state.KeepRunningBatch(param)) {
+    ParameterizedRoutine(param);
+  }
+}
+BENCHMARK(BM_ParameterizedRoutine)->Range(1<<10, 1<<18);
+```
+
+The advantage of using `KeepRunningBatch` is that each batch is counted as
+`param` number of iterations. This means the results for different batch
+sizes are directly comparable as a function of the parameter value.
+
+### Very fast benchmarks
+
+When benchmarking very fast operations (less than 3 ns per iteration), use the
```

Comment:
The library supports C++03, and those users will have to use the `KeepRunning` api. Other users should strongly prefer the range loop, given the reduced overhead and simplicity of the call (even with the slightly unusual opaque variable).

I don't know why a user might want to call `State::iterations` inside the loop, but we can check that case and log a warning. we do this for other calls.